substrate/client/transaction-pool/tests/fatp.rs

```diff
@@ -1975,6 +1975,120 @@ fn fatp_future_is_pruned_by_conflicting_tags() {
 	assert_pool_status!(header02.hash(), &pool, 0, 0);
 }
 
+#[test]
+fn fatp_prune_based_on_inactive_views_tags() {
+	sp_tracing::try_init_simple();
+
+	let (pool, api, _) = pool();
+
+	let xt0 = uxt(Alice, 200);
+	let xt1 = uxt(Alice, 201);
+	let xt2 = uxt(Alice, 202);
+	let xt3 = uxt(Alice, 203);
+	let xt4 = uxt(Alice, 204);
+	let xt5 = uxt(Alice, 205);
+	let xt6 = uxt(Alice, 206);
+	let xt7 = uxt(Alice, 207);
+	let xt8 = uxt(Alice, 208);
+	let xt9 = uxt(Alice, 209);
+	let xt10 = uxt(Alice, 210);
+	let xt11 = uxt(Alice, 211);
+
+	// Push an empty common block.
+	let header01 = api.push_block(1, vec![], true);
+	let event = new_best_block_event(&pool, None, header01.hash());
+	block_on(pool.maintain(event));
+	assert_pool_status!(header01.hash(), &pool, 0, 0);
+	assert_eq!(api.validation_requests().len(), 0);
+
+	// Submit a tx to the txpool.
+	let _ = block_on(pool.submit_and_watch(invalid_hash(), SOURCE, xt1.clone())).unwrap();
+	let _ = block_on(pool.submit_and_watch(invalid_hash(), SOURCE, xt2.clone())).unwrap();
+	assert_pool_status!(header01.hash(), &pool, 0, 2);
+	assert_eq!(api.validation_requests().len(), 2);
+
+	// Push the first retracted fork block, with the ready tx.
+	let header02 = api.push_block(2, vec![xt0.clone()], true);
```

Comment:
please use `push_block_with parent`, I was under impression that push_block does not work well.

Reply:
Also please call forks: `header02a`, `header02b`
At some point I started to use this convention - it is a bit more readable.

Reply:
resolved here: [cf972e8](https://github.com/paritytech/polkadot-sdk/pull/7980/commits/cf972e87cbb5e00b650f44f420f49435c4e400ee).